gnomad_qc/v3/create_release/sanity_checks.py

```diff
@@ -0,0 +1,711 @@
+    :param delimiter: String to use as delimiter when making group label combinations.
+    :return: None
+    """
+
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    if subset != "":
+        subset += delimiter
+
+    label_combos = make_label_combos(label_groups, label_delimiter=delimiter)
+    combo_AC = [t.info[f"AC-{subset}{x}"] for x in label_combos]
+    combo_AN = [t.info[f"AN-{subset}{x}"] for x in label_combos]
+    combo_nhomalt = [t.info[f"nhomalt-{subset}{x}"] for x in label_combos]
+
+    group = label_groups.pop("group")[0]
+    alt_groups = delimiter.join(
+        sorted(label_groups.keys(), key=lambda x: sort_order.index(x))
+    )
+    group_expr = f"{subset}{group}{delimiter}{alt_groups}"
+    annot_dict = {
+        f"sum_AC-{group_expr}": hl.sum(combo_AC),
+        f"sum_AN-{group_expr}": hl.sum(combo_AN),
+        f"sum_nhomalt-{group_expr}": hl.sum(combo_nhomalt),
+    }
+
+    t = t.annotate(**annot_dict)
+
+    for subfield in ["AC", "AN", "nhomalt"]:
+
+        group_expr = f"{subfield}{delimiter}{subset}{group}"
+        alt_groups_expr = f"{subfield}{delimiter}{subset}{group}{delimiter}{alt_groups}"
+
+        generic_field_check(
+            t,
+            (t.info[f"{group_expr}"] != t[f"sum_{alt_groups_expr}"]),
+            f"{group_expr} = sum({alt_groups_expr})",
+            [f"info.{group_expr}", f"sum_{alt_groups_expr}",],
+            verbose,
+        )
+
+
+def sample_sum_sanity_checks(
+    t: Union[hl.MatrixTable, hl.Table],
+    subsets: List[str],
+    info_metrics: List[str],
+    verbose: bool,
+    pops: List[str] = POPS,
+    sexes: List[str] = SEXES,
+) -> None:
+    """
+    Compute afresh the sum of annotations for a specified group of annotations, and compare to the annotated version;
+    displays results from checking the sum of the specified annotations in the terminal.
+    Also check that annotations for all expected sample populations are present.
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Table.
+    :param List[str] subsets: List of sample subsets.
+    :param List[str] info_metrics: List of metrics in info struct of input Table.
+    :param bool verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :param pop_names: Dict with global population names (keys) and population descriptions (values).
+    :return: None
+    :rtype: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+    # Add "" for sum checks on entire callset
+    subsets.append("")
+    # Perform sample sum checks per subset
+    for subset in subsets:
+        pop_names = pops
+        if subset == "hgdp":
+            pop_names = HGDP_POPS
+        if subset == "tgp":
+            pop_names = KG_POPS
+
+        sample_sum_check(t, subset, dict(group=["adj"], pop=pop_names), verbose)
```

Comment:
Does it just use `adj` because both groupings would return the same? I guess I just want to confirm there is no reason to allow for group (or multiple groups) to be passed as an argument?

Reply:
Sounds great, can you add it as a note?